tests/TESTS/LinkLoss/device/source/main.cpp

```diff
@@ -0,0 +1,167 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ble/BLE.h"
+#include "ble/gap/Gap.h"
+#include "ble-service-link-loss/LinkLossService.h"

```

Comment:
@AGlass0fMilk Seeing that type of file reinforce my belief on having a BLE framework to handle most of the boiler plate.

Reply:
there is a process in utils, all it needs is changing the event handler to a chainable one and you can use it